packages/settings/test/settings-spec.ts

```diff
@@ -0,0 +1,176 @@
+import {
+  resetSettings,
+  getSettingsForDocument,
+  getDefaultSettings,
+  setGlobalSettings,
+  setSettingsForDocument,
+  hasSettingsForDocument,
+  clearDocumentSettings,
+  clearSettings,
+} from "../src/settings";
+import { expect } from "chai";
+
+context("settings utilities", () => {
```

Comment:
we invested effort in integrating deepFreeze.
Should we also create a test that the (default) settings are immutable?
Maybe also other kinds of settings (per doc/global)

Reply:
Done for default and global.
It's harder for document because we get a promise. It can be done in the get function but that seems a bit too much (it will probably be called many more times than the set).